### DISCUSSION TITLE
Fresh installation problem on Shelly 1

### DISCUSSION THREAD

**Discussion by Benji1992-HU** [RavenSystem/esp-homekit-devices#2188] [General]
Hello Everyone!
First thing first I want to say thank you for the developers, the project is amazing and I have 3 devices(2x Shelly 1, Shelly rgbw2) which are working more than 3 years.
Currently I want to make a fresh install on a Shelly 1 v3 device, but somewhere the process is fail and I can't add the accessory to my apple home.

**First try:**
The new device contained the original shelly firmware what I updated to the latest version. Then I followed the installation instruction and updated to tasmota with the following command:
`http://192.168.1.15/ota?url=http://ota.tasmota.com/tasmota/shelly/mg2tasmota-Shelly1.zip`
After some minutes the process is finished and the tasmota wifi AP is appeard. I connected to it and made an another firmware update with the following compressed file:
`https://github.com/RavenSystem/haa/releases/latest/download/tasmota2haa.bin.gz`
This also worked well, the HAA wifi AP appeard and I connected to it. And now this is the point where I can't move on.
After I entered my wifi settings the device sayd OK and apparently connected to my network.
It can be confirmed with the device ping:
![image](https://github.com/RavenSystem/esp-homekit-devices/assets/47895802/a10bbd9b-ce1e-4eeb-9d0c-5f4f0d318499)
I waited much time (first try: 10 minute, second: more than 1 hour), but nothing happened. 
The device is connected to my network, but I can't access to the setup mode via `192.168.1.15:4567` and I can't add to my apple home.

**Second try:**
After the update from original Shelly fw is not worked, I tried a wired installation with esptool.
I used the following commands:

1. `esptool.py -p COM3 -b 115200 erase_flash`
2. `esptool.py -p COM3 -b 115200 --before=default_reset --after=hard_reset write_flash -fs 1MB -fm dout 0x0 fullhaaboot.bin`
As I see the commands finished succesfully:
![image](https://github.com/RavenSystem/esp-homekit-devices/assets/47895802/d6deb3ea-9edb-4a02-a09f-7b6f1cc108a2)
![image](https://github.com/RavenSystem/esp-homekit-devices/assets/47895802/db4d65b0-1cfa-4eca-a798-507a82d1e012)

**The final result is the same as in the first try:** 
The device is connected to my network, but I can't access to the setup mode via `192.168.1.15:4567` and I can't add to my apple home.

I tried with and without entering the MEPLHAA Script but no changes:
`{"c":{"io":[[[4],2],[[5],6,0,1]]},"a":[{"0":{"r":[{"g":4}]},"1":{"r":[{"g":4,"v":1}]},"b":[{"g":5},{"g":5,"t":0}]}]}`

I also tried to see the HAA OTA logs, but unfortunately I don't have macOS and GNU/Linux computer.
After some google search I found it is possible to do on Windows PC also, but doesn't worked for me.
The original command:
`nc -kulnw0 45678`
What I tried:
`ncat -lv 45678`
![image](https://github.com/RavenSystem/esp-homekit-devices/assets/47895802/e907da10-ed98-41e1-aa1f-0c44c3310fdd)

Have somebody tried this on Windows PC?

Can somebody help me what I make wrong?

**Comment by RavenSystem** [comment]
Very strange. Maybe HAA logs are not working under Windows because firewall. Have you HAA Home Manager App?

**Reply by Benji1992-HU** [reply]
Yes I have, but doesn't show anything. When I am updating some of my already installed devices int the HAA Home Manager App I can see the whole process, so I think the app is working well.

**Reply by RavenSystem** [reply]
Can you try with new fullhaaboot released just now?

**Reply by Benji1992-HU** [reply]
Yes I try and give you a feedback.

**Reply by Benji1992-HU** [reply]
Now the HAA Home Manager App is working. It's looks like the installation is freezing at otmani.bin.sec download.
Here is the log content:
```
Adv Log (c) 2022-2024 José A. Jiménez Campos

0010.650 192.168.1.15-2E5001: HAA Installer 7.8.2
0010.651 192.168.1.15-2E5001: REPO https://github.com/RavenSystem/haa/releases/latest/download 443
0010.651 192.168.1.15-2E5001: **** HAABOOT
0010.953 192.168.1.15-2E5001: INIT
0012.949 192.168.1.15-2E5001: >>> Get sign
0012.950 192.168.1.15-2E5001: *** DOWNLOADING
0012.951 192.168.1.15-2E5001: *** FORWARDING github.com:443/RavenSystem/haa/releases/latest/download/otamain.bin.sec
0012.951 192.168.1.15-2E5001: *** CONNECT
0013.023 192.168.1.15-2E5001: DNS..OK
0013.024 192.168.1.15-2E5001: Socket..OK
0019.694 192.168.1.15-2E5001: SNI..OK
0019.695 192.168.1.15-2E5001: SSLConnect..OK
0019.703 192.168.1.15-2E5001: Write..OK
0021.698 192.168.1.15-2E5001: ! Read -1
0021.699 192.168.1.15-2E5001: ! SSL 2
0021.701 192.168.1.15-2E5001: *** FORWARDING github.com:443/RavenSystem/haa/releases/latest/download/otamain.bin.sec
0021.702 192.168.1.15-2E5001: *** CONNECT
0021.703 192.168.1.15-2E5001: DNS..OK
0021.703 192.168.1.15-2E5001: Socket..OK
0021.761 192.168.1.15-2E5001: Connect..OK
0021.762 192.168.1.15-2E5001: SSL..OK
0021.762 192.168.1.15-2E5001: FD..OK
```
Second try:
```
Adv Log (c) 2022-2024 José A. Jiménez Campos

0010.659 192.168.1.15-2E5001: HAA Installer 7.8.2
0010.660 192.168.1.15-2E5001: REPO https://github.com/RavenSystem/haa/releases/latest/download 443
0010.660 192.168.1.15-2E5001: **** HAABOOT
0012.958 192.168.1.15-2E5001: >>> Get sign
0012.959 192.168.1.15-2E5001: *** DOWNLOADING
0012.960 192.168.1.15-2E5001: *** FORWARDING github.com:443/RavenSystem/haa/releases/latest/download/otamain.bin.sec
0012.960 192.168.1.15-2E5001: *** CONNECT
0013.004 192.168.1.15-2E5001: DNS..OK
0013.005 192.168.1.15-2E5001: Socket..OK
0013.045 192.168.1.15-2E5001: Connect..OK
0013.047 192.168.1.15-2E5001: SSL..OK
0013.047 192.168.1.15-2E5001: FD..OK
0028.126 192.168.1.15-2E5001: SNI..OK
0028.127 192.168.1.15-2E5001: SSLConnect..! -1
0028.127 192.168.1.15-2E5001: ! SSL -308
0033.127 192.168.1.15-2E5001: *** FORWARDING github.com:443/RavenSystem/haa/releases/latest/download/otamain.bin.sec
0033.128 192.168.1.15-2E5001: *** CONNECT
0033.128 192.168.1.15-2E5001: DNS..OK
0033.129 192.168.1.15-2E5001: Socket..OK
0033.169 192.168.1.15-2E5001: Connect..OK
0033.171 192.168.1.15-2E5001: SSL..OK
0033.171 192.168.1.15-2E5001: FD..OK
0039.682 192.168.1.15-2E5001: SNI..OK
0039.683 192.168.1.15-2E5001: SSLConnect..OK
0039.691 192.168.1.15-2E5001: Write..OK

0042.847 192.168.1.15-2E5001: Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
0042.853 192.168.1.15-2E5001: X-Frame-Options: deny
0042.855 192.168.1.15-2E5001: X-Content-Type-Options: nosniff
0042.858 192.168.1.15-2E5001: X-XSS-Protection: 0
0042.859 192.168.1.15-2E5001: Referrer-Policy: no-referrer-when-downgrade
0042.863 192.168.1.15-2E5001: Content-Security-Policy: default-src 'none'; base-uri 'self'; child-src github.com/assets-cdn/worker/ gist.github.com/assets-cdn/worker/; connect-src 'self' uploads.github.com www.githubstatus.com collector.github.com raw.githubusercontent.com api.github.com github-cloud.s3.amazonaws.com github-production-repository-file-5c1aeb.s3.amazonaws.com github-production-upload-manifest-file-7fdce7.s3.amazonaws.com github-production-user-asset-6210df.s3.amazonaws.com cdn.optimizely.com logx.optimizely.com/v1/events api.githubcopilot.com objects-origin.githubusercontent.com *.actions.githubusercontent.com wss://*.actions.githubusercontent.com productionresultssa0.blob.core.windows.net/ productionresultssa1.blob.core.windows.net/ productionresultssa2.blob.core.windows.net/ productionresultssa3.blob.core.windows.net/ productionresultssa4.blob.core.windows.net/ productionresultssa5.blob.core.windows.net/ productionresultssa6.blob.core.windows.net/ productionresultssa7.blob.core.windows.net/ productionresultssa8.blob.core.windows.net/ productionresultssa9.blob.core.windows.net/ github-production-repository-image-32fea6.s3.amazonaws.com github-production-release-asset-2e65be.s3.amazonaws.com insights.github.com wss://alive.github.com; font-src github.githubassets.com; form-action 'self' github.com gist.github.com objects-origin.githubusercontent.com; frame-ancestors 'none'; frame-src viewscreen.githubusercontent.com notebooks.githubusercontent.com support.github.com; img-src 'self' data: github.githubassets.com media.githubusercontent.com camo.githubusercontent.com identicons.github.com avatars.githubusercontent.com github-cloud.s3.amazonaws.com objects.githubusercontent.com secured-user-images.githubusercontent.com/ user-images.githubusercontent.com/ private-user-images.githubusercontent.com opengraph.githubassets.com github-production-user-asset-6210df.s3.amazonaws.com customer-stories-feed.github.com spotlights-feed.github.com objects-origin.githubusercontent.com *.githubusercontent.com; manifest-src 'self'; media-src github.com user-images.githubusercontent.com/ secured-user-images.githubusercontent.com/ private-user-images.githubusercontent.com github-production-user-asset-6210df.s3.amazonaws.com gist.github.com; script-src github.githubassets.com; style-src 'unsafe-inline' github.githubassets.com; upgrade-insecure-requests; worker-src github.com/assets-cdn/worker/ gist.github.com/assets-cdn/worker/
0043.075 192.168.1.15-2E5001: Set-Cookie: _gh_sess=pgVdfRxdJumgpzI41fjG68rNSCoKlApC37a%2FQMA6w81dWUeLbgF9eFHdRpuvWRvnoGulaLapSkrT7Je9c2P7JKLDcwQwo7S%2BwqVtsw3gIN5tbMhy5d5GVKHeLJzwYoCqfIvdfRAhAJhgRY8pWzNteOIdq6ukVGCgwfi49QY1z5qLwVuG9DkQNKPfx8cirnnDA%2FwVNWx3B9vYyoE9VQjiEMpvz%2FnsRlAvBHHeCucMiMbjNEnZT89Ifj0qTo0cPIcQm42DYmO%2ByLBK94ssG%2Bza2g%3D%3D--ajmgcmue4fyJo4xz--PMxZKFrPkR2%2F34M2L1aoPA%3D%3D; Path=/; HttpOnly; Secure; SameSite=Lax
0043.110 192.168.1.15-2E5001: Set-Cookie: _octo=GH1.1.907110997.1704973182; Path=/; Domain=github.com; Expires=Sat, 11 Jan 2025 11:39:42 GMT; Secure; SameSite=Lax
0043.122 192.168.1.15-2E5001: Set-Cookie: logged_in=no; Path=/; Domain=github.com; Expires=Sat, 11 Jan 2025 11:39:42 GMT; HttpOnly; Secure; SameSite=Lax
0043.132 192.168.1.15-2E5001: Content-Length: 0
0043.134 192.168.1.15-2E5001: X-GitHub-Request-Id: E6E5:26964F:F879CA8:FB04C88:659FD377

0043.139 192.168.1.15-2E5001: 

0043.140 192.168.1.15-2E5001: 

0043.140 192.168.1.15-2E5001: 
0043.141 192.168.1.15-2E5001: HTTP returns 302

0043.141 192.168.1.15-2E5001: 
0043.166 192.168.1.15-2E5001: *** FORWARDING github.com:443/RavenSystem/haa/releases/download/12.10.3/otamain.bin.sec
0043.166 192.168.1.15-2E5001: *** CONNECT
0043.167 192.168.1.15-2E5001: DNS..OK
0043.168 192.168.1.15-2E5001: Socket..OK
0043.263 192.168.1.15-2E5001: Connect..OK
0043.265 192.168.1.15-2E5001: SSL..OK
0043.265 192.168.1.15-2E5001: FD..OK
```

**Reply by RavenSystem** [reply]
Are you powering Shelly with mains or with USB?

**Reply by Benji1992-HU** [reply]
After the wired programming I am using mains(230V).

**Reply by RavenSystem** [reply]
It seems like a WiFi issue or a power issue. Freezing is not something normal.

**Reply by Benji1992-HU** [reply]
I tried to update some of from my old devices (from 12.10.2)and it’s looks like the installation is also freezing but at different point.
[Jan 11 130453 HAA Log.log](https://github.com/RavenSystem/esp-homekit-devices/files/13902912/Jan.11.130453.HAA.Log.log)

**Reply by RavenSystem** [reply]
It seems that something is wrong with your WiFi (router or AP). Installer used currently by 12.10.2 is different that new installer used by your new Shelly. However, installer from 12.10.2 was used to install it when you updated HAA some days ago, and all went fine. Try to reboot your WiFi routers and APs.

**Reply by RavenSystem** [reply]
FYI: I updated all my HAA devices (about 50 devices) from 12.10.2 to 12.10.3 just now, and all went fine.

**Reply by Benji1992-HU** [reply]
Hm, the old device update is finished successfully, maybe in this case only the log was freezed. 
It is possible to do the fresh install completely offline via esp tool?
